### PR TITLE
Chore: Cleanup usageStatsMock duplication

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:
Removes duplicated usageStatsMock setup as suggested https://github.com/grafana/grafana/pull/39558#issuecomment-925886561 to keep things clean.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func setupTestEnv(t testing.TB) *OSSAccessControlService {
 
 	cfg := setting.NewCfg()
 	cfg.FeatureToggles = map[string]bool{"accesscontrol": true}
-	ac := ProvideService(cfg, &usageStatsMock{metricsFuncs: make([]usagestats.MetricsFunc, 0)})
+	ac := ProvideService(cfg, &usagestats.UsageStatsMock{MetricsFuncs: make([]usagestats.MetricsFunc, 0)})
```
Comment (DanCech) [inline pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol_test.go:22]:
```suggestion
	ac := ProvideService(cfg, &usagestats.UsageStatsMock{T: t, MetricsFuncs: make([]usagestats.MetricsFunc, 0)})
```

Comment (undef1nd) [reply pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol_test.go:22]: thanks, I think I need to remove `make([]usagestats.MetricsFunc, 0)` basically everywhere where we use this mock.

Comment (DanCech) [reply pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol_test.go:22]: yeah afaik it's not needed anywhere which will keep things nice and clean :)

**Review by DanCech — APPROVED** [review]
LGTM

**Pull request merged**